docs/sphinx/using/hardware.rst

```diff
@@ -0,0 +1,96 @@
+CUDA Quantum Hardware Backends
+*********************************
+
+The hardware vendors currently available in CUDA Quantum are as follows.
+
+IonQ
+==================================
+
+Setting Credentials
+###################
+
+To execute programs on IonQ hardware from either C++ or Python, CUDA Quantum
+will look for an API key stored in the `IONQ_API_KEY` environment variable. 
+This may be set as follows, replacing the string with the API key generated
+from your `IonQ account <https://cloud.ionq.com/>`_.
+
+.. code:: bash
+
+  export IONQ_API_KEY="ionq_generated_api_key"
+
+For C++, it's straightforward to control the target QPU via the `--target`
+argument to `nvq++`. This will look for the `IONQ_API_KEY` in your environment,
+authenticate it with the IonQ API, and submit any quantum kernel executions within
+the file to the hardware.
+
+.. code:: bash 
+
+    nvq++ --target ionq src.cpp ...
+
+In python, the target may be controlled with the `cudaq.set_target()` [TODO: LINK]
+function. This will set the target for any kernel executions within the file, 
+and will go through the same credential scheme as discussed in the C++ case. 
+
+.. code:: python 
+
+    cudaq.set_target('ionq')
+
+Specifying the QPU
+###################
+
+By default, the IonQ target will use the :code:`simulator` QPU.
+To specify which IonQ QPU to use, set the :code:`qpu` parameter.
+A list of available QPUs can be found `in the API documentation <https://docs.ionq.com/#tag/jobs>`_.
+
+.. code:: c
+
+    auto &platform = cudaq::get_platform();
+    platform.setTargetBackend("ionq;qpu;qpu.aria-1");
```

Comment:
I don't think we want to expose this in this way in C++. I would prefer we follow the quantinuum.config target file and expose a way to specify this at the command line (`--ionq-backend aria-1`). 

If we do allow someone to set this at runtime, we should add a setTargetBackend overload that takes a std::map so users don't have to think about how to structure this string.